detector/tracker/SiTrackerEndcap_o2_v01ext_geo.cpp

```diff
@@ -173,7 +175,8 @@ static Ref_t create_detector(Detector& theDetector, xml_h e, SensitiveDetector s
     thisLayer.zPosition = sumZ/ring_num; // average z
     thisLayer.distanceSensitive = innerR;
     thisLayer.lengthSensitive = outerR - innerR;
-    thisLayer.petalNumber = ring_num; // number of rings in petalNumber, needed for tracking
+    thisLayer.petalNumber = petal_num;
```

Comment:
Why not set this to `nmodules` that is fixed? `petal_num` is being set to `nmodules` `nmodules` times

Reply:
`nmodules` is no longer in scope for using it when `thisLayer.petalNumber` is set. However, I can setting it `nmodules` times is unnecessary, and I can lift it out of the loop.